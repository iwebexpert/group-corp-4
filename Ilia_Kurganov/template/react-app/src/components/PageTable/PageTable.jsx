import React, { useEffect, useState } from 'react'
import './PageTable.scss'
import PropTypes from 'prop-types'
import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'
import Delete from '@mui/icons-material/Delete'
import Edit from '@mui/icons-material/Edit'
import SaveAltIcon from '@mui/icons-material/SaveAlt'
import { TextField } from '@mui/material'

const PageTable = ({ tableRows, delitePage, role }) => {
  const newTableRow2 = () => {
    return tableRows.map((item) => {
      return { ...item, isEdit: true }
    })
  }

  const [newTableRow, setNewTableRow] = useState(newTableRow2)

  useEffect(() => {
    setNewTableRow(newTableRow2)
  }, [tableRows])

  const editRow = (key, action) => {
    setNewTableRow((row) => {
      return row.map((item) => {
        if (item.id === key) {
          return { ...item, isEdit: action === 'save' ? true : false }
        }
        return item
      })
    })
  }

  const changeField = (key, field, value) => {
    setNewTableRow((row) =>
      row.map((item) => {
        if (item.id === key) {
          const f = { ...item }
          f[field] = value
          return f
        }
        return item
      }),
    )
  }

  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Id</TableCell>
            <TableCell align="right">URL</TableCell>
            <TableCell align="right">Title</TableCell>
            <TableCell align="right">Content</TableCell>
            <TableCell align="right">UserId</TableCell>
            {role === 'admin' ? <TableCell align="right">&nbsp;</TableCell> : null}
          </TableRow>
        </TableHead>
        <TableBody>
          {newTableRow.map((page, i) => (
            <TableRow key={page.id} sx={{ '&:last-child td, &:last-child th': { border: 0 } }}>
              <TableCell component="th" scope="row">
                {i + 1}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  page.url
                ) : (
                  <TextField
                    id="url"
                    value={page.url}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  page.title
                ) : (
                  <TextField
                    id="title"
                    value={page.title}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  page.content
                ) : (
                  <TextField
                    id="content"
                    value={page.content}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  page.userId
                ) : (
                  <TextField
                    id="userId"
                    value={page.userId}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                )}
              </TableCell>
              {role === 'admin' ? (
                <TableCell align="center">
                  <Delete
                    sx={{ cursor: 'pointer', color: 'red', ml: 2 }}
                    onClick={() => delitePage(page.id)}
                  ></Delete>
                  {page.isEdit ? (
                    <Edit
                      sx={{ cursor: 'pointer', color: 'blue', ml: 4 }}
                      onClick={() => editRow(page.id, 'edit')}
                    ></Edit>
                  ) : (
                    <SaveAltIcon
                      sx={{ cursor: 'pointer', color: 'green', ml: 4 }}
                      onClick={() => editRow(page.id, 'save')}
                    ></SaveAltIcon>
                  )}
                </TableCell>
              ) : null}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}

PageTable.propTypes = {
  tableRows: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      userId: PropTypes.number.isRequired,
    }),
  ).isRequired,
  delitePage: PropTypes.func.isRequired,
  role: PropTypes.string,
}

export default PageTable
