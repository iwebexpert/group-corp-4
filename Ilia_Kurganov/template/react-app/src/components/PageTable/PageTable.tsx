import React from 'react'
import { Link } from 'react-router-dom'
import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'
import Delete from '@mui/icons-material/Delete'
import Edit from '@mui/icons-material/Edit'
import SaveAltIcon from '@mui/icons-material/SaveAlt'
import { TextField } from '@mui/material'
import { PagePayload } from 'src/actions/actionsPages'

type PageTableType = {
  tableRows: PagePayload[]
  delitePage: (id: string) => void
  editPage: (id: string) => void
  savePage: (id: string) => void
  changeField: (id: string, field: string, value: string) => void
  role: string
}

const PageTable = ({
  tableRows,
  delitePage,
  editPage,
  savePage,
  changeField,
  role,
}: PageTableType) => {
  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Id</TableCell>
            <TableCell align="right">URL</TableCell>
            <TableCell align="right">Title</TableCell>
            <TableCell align="right">Content</TableCell>
            <TableCell align="right">UserId</TableCell>
            {role === 'admin' ? <TableCell align="right">&nbsp;</TableCell> : null}
          </TableRow>
        </TableHead>
        <TableBody>
          {tableRows.map((page, i) => (
            <TableRow key={page.id} sx={{ '&:last-child td, &:last-child th': { border: 0 } }}>
              <TableCell component="th" scope="row">
                <Link to={`${page.id}`}>{i + 1}</Link>
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  <TextField
                    id="url"
                    value={page.url}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                ) : (
                  page.url
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  <TextField
                    id="title"
                    value={page.title}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                ) : (
                  page.title
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  <TextField
                    id="content"
                    value={page.content}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                ) : (
                  page.content
                )}
              </TableCell>
              <TableCell align="right">
                {page.isEdit ? (
                  <TextField
                    id="userId"
                    value={page.userId}
                    variant="outlined"
                    onChange={(e) => changeField(page.id, e.target.id, e.target.value)}
                  />
                ) : (
                  page.userId
                )}
              </TableCell>
              {role === 'admin' ? (
                <TableCell align="center">
                  <Delete
                    className="delete-svg"
                    id="edit-svg-for-jest-only"
                    sx={{ cursor: 'pointer', color: 'red', ml: 2 }}
                    onClick={() => delitePage(page.id)}
                  ></Delete>
                  {page.isEdit ? (
                    <SaveAltIcon
                      className="save-svg"
                      sx={{ cursor: 'pointer', color: 'green', ml: 4 }}
                      onClick={() => savePage(page.id)}
                    ></SaveAltIcon>
                  ) : (
                    <Edit
                      className="edit-svg"
                      sx={{ cursor: 'pointer', color: 'blue', ml: 4 }}
                      onClick={() => editPage(page.id)}
                    ></Edit>
                  )}
                </TableCell>
              ) : null}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}

export default PageTable
