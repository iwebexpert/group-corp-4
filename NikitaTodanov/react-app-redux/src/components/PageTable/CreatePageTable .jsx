import React from "react";
import PropTypes from "prop-types";

import Typography from "@mui/material/Typography";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import DeleteIcon from "@mui/icons-material/Delete";
import EditIcon from "@mui/icons-material/Edit";

import styled from "@emotion/styled";

const DeleteIconCustom = styled(DeleteIcon)`
  color: blue;
  opacity: 0.5:
  cursor: pointer;
  margin-left:10px;
`;

const EditIconCustom = styled(EditIcon)`
  color: green;
  cursor: pointer;
  margin-left: 10px;
`;

export function CreatePageTable({
  pages,
  onDeletePage,
  onEditPageMode,
  roles,
}) {
  return (
    <>
      <Typography component="h2" color="primary">
        Созданные страницы
      </Typography>
      <Table size="small">
        <TableHead>
          <TableRow>
            <TableCell>#</TableCell>
            <TableCell>URL</TableCell>
            <TableCell>Наименование</TableCell>
            <TableCell>Комментарий</TableCell>
            <TableCell>UserID</TableCell>
            <TableCell>Изменить</TableCell>
            <TableCell>&nbsp;</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {pages.map((obj, index) => (
            <TableRow key={index}>
              <TableCell>{index + 1}</TableCell>
              <TableCell>{obj.url}</TableCell>
              <TableCell>{obj.title}</TableCell>
              <TableCell>{obj.content}</TableCell>
              <TableCell>{obj.userId.toString()}</TableCell>
              <TableCell>
                <EditIconCustom
                  color="secondary"
                  onClick={() => onEditPageMode(obj.id)}
                />
                <DeleteIconCustom
                  color="secondary"
                  onClick={() => onDeletePage(obj.id)}
                />
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </>
  );
}

CreatePageTable.defaultProps = {
  equipment: [],
  onDeleteEquipment: () => {},
};

CreatePageTable.propTypes = {
  equipment: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      url: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      title: PropTypes.string,
      content: PropTypes.string,
      userId: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
        .isRequired,
    })
  ),
  onDeleteEquipment: PropTypes.func.isRequired,
};
