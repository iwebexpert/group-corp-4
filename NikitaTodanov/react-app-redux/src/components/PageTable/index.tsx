import CreatePageTable from './CreatePageTable '


export default CreatePageTable 
