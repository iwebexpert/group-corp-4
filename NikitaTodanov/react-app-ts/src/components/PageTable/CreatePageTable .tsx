import React from "react";



import Typography from "@mui/material/Typography";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import DeleteIcon from "@mui/icons-material/Delete";
import EditIcon from "@mui/icons-material/Edit";
import AddIcon from "@mui/icons-material/Add";

import styled from "@emotion/styled";

const DeleteIconCustom = styled(DeleteIcon)`
  color: blue;
  opacity: 0.5:
  cursor: pointer;

`;

const EditIconCustom = styled(EditIcon)`
  color: green;
  cursor: pointer;
`;

type CreatePageItem = {
  id: string,
  url: string,
  title: string,
  content: string,
  userId: string
}

type CreatePageTableProps = {
  pages: CreatePageItem[],
  onDeletePage: (id: string) => void
  onEditPageMode: (id: string) => void
}

const CreatePageTable: React.FC<CreatePageTableProps> = ({ pages, onDeletePage, onEditPageMode }) => {
  return (
    <>
      <Typography component="h2" color="primary">
        Созданные страницы
      </Typography>
      <Table size="small">
        <TableHead>
          <TableRow>
            <TableCell>#</TableCell>
            <TableCell>URL</TableCell>
            <TableCell>Наименование</TableCell>
            <TableCell>Контент</TableCell>
            <TableCell>UserID</TableCell>
            <TableCell>Изменить</TableCell>
            <TableCell>&nbsp;</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {pages.map((obj, index) => (
            <React.Fragment key={index}>
              <TableRow>
                <TableCell>{index + 1}</TableCell>
                <TableCell>{obj.url}</TableCell>
                <TableCell>{obj.title}</TableCell>
                <TableCell>{obj.content}</TableCell>
                <TableCell>{obj.userId.toString()}</TableCell>
                <TableCell>
                  <EditIconCustom
                    color="secondary"
                    onClick={() => onEditPageMode(obj.id)}
                  />
                  <DeleteIconCustom
                    color="secondary"
                    onClick={() => onDeletePage(obj.id)}
                  />
                </TableCell>
              </TableRow>
            </React.Fragment>
          ))}
        </TableBody>
      </Table>
    </>
  );
}

export default CreatePageTable