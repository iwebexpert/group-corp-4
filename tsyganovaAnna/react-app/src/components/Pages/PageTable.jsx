import React from 'react'
import PropTypes from 'prop-types'

import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import IconButton from '@mui/material/IconButton'
import EditIcon from '@mui/icons-material/Edit'
import DeleteIcon from '@mui/icons-material/Delete'

export default function PageTable({ pages, showButton, onEditPage, onDeletePage }) {
  return (
    <TableContainer sx={{ maxHeight: 440 }}>
      <Table stickyHeader aria-label="sticky table" size="small">
        <TableHead>
          <TableRow>
            <TableCell>â„–</TableCell>
            <TableCell>Url</TableCell>
            <TableCell>Title</TableCell>
            <TableCell>Content</TableCell>
            {showButton && (
              <>
                <TableCell>&nbsp;</TableCell>
                <TableCell>&nbsp;</TableCell>
              </>
            )}
          </TableRow>
        </TableHead>
        <TableBody>
          {pages.map((tabelItem, index) => (
            <TableRow key={index}>
              <TableCell>{index + 1}</TableCell>
              <TableCell>{tabelItem.url}</TableCell>
              <TableCell>{tabelItem.title}</TableCell>
              <TableCell>{tabelItem.content}</TableCell>
              {showButton && (
                <>
                  <TableCell>
                    <IconButton onClick={() => onEditPage(tabelItem.id)} aria-label="edit">
                      <EditIcon />
                    </IconButton>
                  </TableCell>
                  <TableCell>
                    <IconButton onClick={() => onDeletePage(tabelItem.id)} aria-label="delete">
                      <DeleteIcon />
                    </IconButton>
                  </TableCell>
                </>
              )}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}
PageTable.defaultProps = {
  pages: [],
  showButton: false,
  onEditPage: () => {},
  onDeletePage: () => {},
}

PageTable.propTypes = {
  pages: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired || PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      content: PropTypes.string,
    }),
  ),
  showButton: PropTypes.bool.isRequired,
  onEditPage: PropTypes.func.isRequired,
  onDeletePage: PropTypes.func.isRequired,
}
