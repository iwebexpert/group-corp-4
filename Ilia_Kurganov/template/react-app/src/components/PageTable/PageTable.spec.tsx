import React from 'react'
import * as renderer from 'react-test-renderer'
import { test, expect } from '@jest/globals'
import { shallow, mount } from 'enzyme'
import PageTable from './PageTable'
import { fn } from 'jest-mock'
import { MemoryRouter } from 'react-router-dom'
import { tableRowClasses } from '@mui/material'

test('PageTable have same snapshot', () => {
  const row = [
    {
      id: '442423423423',
      url: 'page2',
      title: 'Page2',
      content: 'Some content...',
      userId: '3',
    },
    {
      id: '321312',
      url: 'page2',
      title: 'Page2',
      content: 'Some content...',
      userId: '4',
    },
  ]
  const mockDelitePage = fn()
  const mockEditPage = fn()
  const mockSavePage = fn()
  const mockChangeField = fn()
  const table = renderer.create(
    <MemoryRouter>
      <PageTable
        tableRows={row}
        delitePage={mockDelitePage}
        editPage={mockEditPage}
        savePage={mockSavePage}
        changeField={mockChangeField}
        role="admin"
      />
    </MemoryRouter>,
  )
  const treeJson = table.toJSON()
  expect(treeJson).toMatchSnapshot()
})

test('PageTable should be rendered', () => {
  const row = [
    {
      id: '442423423423',
      url: 'page2',
      title: 'Page2',
      content: 'Some content...',
      userId: '3',
      isEdit:true
    },
  ]
  const mockDelitePage = fn()
  const mockEditPage = fn()
  const mockSavePage = fn()
  const mockChangeField = fn()
  const table = mount(
    <MemoryRouter>
      <PageTable
        tableRows={row}
        delitePage={mockDelitePage}
        editPage={mockEditPage}
        savePage={mockSavePage}
        changeField={mockChangeField}
        role="admin"
      />
    </MemoryRouter>,
  )

  // console.log(table.debug())

    table.find('svg.edit-svg').simulate('click')
    // table.find({type:'svg'}).at(2).simulate('click')
  //   table.find('svg').simulate('click')
  //   expect(mockEditPage.mock.calls.length).toBe(1)
    expect(mockEditPage).toBeCalled()
  //   expect(mockEditPage).toHaveBeenCalled()

  //   expect(mockEditPage.mock.calls[0][0]).toEqual('442423423423')


  table.find("#content").at(0).simulate('change', {target:{value:'12345'}})
  table.find('svg.save-svg').simulate('click')

  expect(mockEditPage.mock.calls[0][0]).toEqual('442423423423')
})
