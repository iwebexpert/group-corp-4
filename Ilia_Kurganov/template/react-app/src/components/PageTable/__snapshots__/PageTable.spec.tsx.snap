// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTable have same snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root css-quj9j6-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Id
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
          scope="col"
        >
          URL
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
          scope="col"
        >
          Title
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
          scope="col"
        >
          Content
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
          scope="col"
        >
          UserId
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
          scope="col"
        >
          Â 
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <a
            href="/442423423423"
            onClick={[Function]}
          >
            1
          </a>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          page2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          Page2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          Some content...
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ujkmd3-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-j7m7gu-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <a
            href="/321312"
            onClick={[Function]}
          >
            2
          </a>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          page2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          Page2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          Some content...
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          4
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ujkmd3-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-j7m7gu-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
